include/kernels/gauge_utils.cuh

```diff
@@ -19,14 +20,16 @@ namespace quda
   // matrix+matrix = 18 floating-point ops
   // => Total number of floating point ops per function call
   // dims * (2*18 + 4*198) = dims*828
-  using computeStapleOps = thread_array<int, 4>;
-  template <typename Arg, typename Staple, typename Int>
-  __host__ __device__ inline void computeStaple(const Arg &arg, const int *x, const Int *X, const int parity, const int nu, Staple &staple, const int dir_ignore)
+  using computeStapleOps = KernelOps<thread_array<int, 4>>;
```

Comment:
I appreciate the idea of declaring this once and using it across multiple files (`gauge_ape.cuh`, etc), but for something this simple, I think we should also just have the classes that currently inherit from `computeStapleOps` just textually inherit from `KernelOps<thread_array<int, 4>>` as well. Maybe I'm missing a particular reason for having this reused abstraction, though.

Reply:
The reason for the abstraction was to keep the list of kernel operations used by a function near that function, so that if the requirements changed, it only needs to be changed in one place, which is right above the function being modified.  Also, it serves as documentation for the main kernel, explaining where the resources are used (which would otherwise not be obvious if many functions are called from the main kernel).  Since the kernels don't change that often, I would agree that the first reason may not matter much, but I think the documentation effect of using an alias for the operations needed for the function calls a kernel makes may make it worthwhile.

Reply:
That makes sense; I agree with your evaluation that it being self-documenting makes it worthwhile here, so I rescind my comment!